src/store/modules/keymap.js

```diff
@@ -374,6 +374,13 @@ const mutations = {
       }
     }
   },
+  duplicateLayer(state, { layer }) {
+    if (state.keymap.length !== MAX_LAYERS) {
```

Comment:
< MAX_LAYERS is probably more appropriate. What happens if a bug causes length to be 17?

Reply:
I think we should also change the state of the button in the UI so that when MAX_LAYERS are filled we should probably disable cloning. Adding cloning introduces other issues now. If you have a sparse keymap with gaps where there are empty layers, the firmware will generate a larger firmware.

You've now opened a new can of worms.

Reply:
I think this cloning solution will work most of the time(don't let perfect be the enemy of good :) ). In the case of gaps generating larger firmware, I don't think that is a problem specific to cloning. You have that issue if you delete layers anyway.

Reply:
Agreed. Just wanted you to be aware of why it might cause your firmware to bloat